metricproviders/prometheus/prometheus.go

```diff
@@ -138,11 +139,25 @@ func (p *Provider) processResponse(metric v1alpha1.Metric, response model.Value)
 }
 
 // NewPrometheusProvider Creates a new Prometheus client
-func NewPrometheusProvider(api v1.API, logCtx log.Entry) *Provider {
-	return &Provider{
+func NewPrometheusProvider(api v1.API, logCtx log.Entry, metric v1alpha1.Metric) (*Provider, error) {
+	provider := &Provider{
 		logCtx: logCtx,
 		api:    api,
 	}
+
+	if metric.Provider.Prometheus == nil || metric.Provider.Prometheus.Timeout == nil {
```

Comment:
I think that we can't end up in a nil pointer situation, because if `metric.Provider.Prometheus == nil` the compiler won't evaluate the next expression which is `metric.Provider.Prometheus.Timeout == nil`. It would pass to the nested block directly. Please correct me if I misunderstood ur point.

Reply:
Ahh yea I see what the logic is I was somehow thinking you where also usin gmetric.Provider inside the block. I am not sure it's actually possible to get metric.Provider.Prometheus == nil inside the function but the check is probably still good